cmake/interrogate_setup_dot_py.py

```diff
@@ -199,6 +198,44 @@ def setup(*args, **kwargs):
         with open(outfile, 'w') as out:
             out.write('\n'.join(result))
 
+        attrs = dict(
+            name=package_name,
+            version=version,
+            package_dir=package_dir,
+            packages=pkgs,
+            scripts=scripts,
+            py_modules=modules,
+            entry_points=entry_points,
+        )
+
+        from setuptools.dist import Distribution
+        dist = Distribution(attrs)
+        dist.script_name = 'setup.py'
+
+        # egg_info command
+        from setuptools.command.egg_info import egg_info
+        cmd_egg_info = egg_info(dist)
+        # overwrite egg_base from package_dir to install_dir
+        # in order to create *.egg-info directory in the install_dir
+        cmd_egg_info.egg_base = install_dir
+        # make sure install directory exists
+        # because egg_info command assumes that
+        if not os.path.exists(install_dir):
+            os.makedirs(install_dir)
+        cmd_egg_info.finalize_options()
+        cmd_egg_info.run()
+
+        # easy_install command
+        import pkg_resources
+        from setuptools.command.easy_install import easy_install
+        cmd_easy_install = easy_install(dist)
+        # overwrite script_dir and not call finalize_options method
+        # to avoid many assertions of easy_install command like check_site_dir
+        cmd_easy_install.script_dir = script_dir
+        metadata = pkg_resources.PathMetadata(cmd_egg_info.egg_base, cmd_egg_info.egg_info)
+        dist_pkg_resources = pkg_resources.Distribution(install_dir, project_name=package_name, metadata=metadata)
+        cmd_easy_install.install_wrapper_scripts(dist_pkg_resources)
```

Comment:
If a package doesn't use `entry_points` what side effects does this block (running `egg_info` and `easy_install`) have? Could it be made conditional to reduce the chance of regressions - only packages which start using entry points would exercise this logic?